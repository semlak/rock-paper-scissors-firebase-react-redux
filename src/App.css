.App {
  text-align: center;
}

.App-link {
  color: #61dafb;
}

.main {
  margin-top: 2em;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
