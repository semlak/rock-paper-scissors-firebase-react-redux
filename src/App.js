import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { connect } from 'react-redux';

import Main from './pages/Main';
// import { Navbar, Landing, requireAuth, AuthenticationModal } from './components';
import { Landing, requireAuth, AuthenticationModal } from './components';
import { NavbarContainer } from './containers';

import { fetchUser } from './actions/user';

export class App extends Component {
  componentDidMount = () => {
    // this.props.fetchUser && typeof this.props.fetchUser === 'function' && this.props.fetchUser();
    this.props.fetchUser();
  }

  render() {
    // console.log('rendering App component');
    // console.log(`Your process.env.PUBLIC_URL: '${process.env.PUBLIC_URL}'`);
    return (
      <BrowserRouter basename={process.env.PUBLIC_URL}>
        <div className="App">
          <NavbarContainer />
          <AuthenticationModal />
          <div className="container main">
            {/* <Route exact path="/" component={SignIn} /> */}
            <Route exact path="/" component={Landing} />
            <Route path="/app" component={requireAuth(Main)} />
          </div>
        </div>
      </BrowserRouter>
    );
  }
}

// export default App;
export default connect(null, { fetchUser })(App);
