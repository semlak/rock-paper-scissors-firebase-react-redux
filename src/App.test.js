import React from 'react';
import ReactDOM from 'react-dom';

// import Enzyme, { shallow } from 'enzyme';
import { shallow } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';




import { App } from './App';

const mockStore = configureStore();

const initialStoreState = {
  auth: null,
  game: {
    gameStatus: "NO_GAME",
  },
  gathering: null,
  modals: {
    authenticationModalOpen: false
  }
};

// Enzyme.configure({ adapter: new Adapter() });
// const fakeStore = {};

describe('components', () => {
  describe('App', () => {
    const mockFetchUserFn = jest.fn();
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<App fetchUser={mockFetchUserFn} />);
    });
    it('shallow renders without crashing', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders without crashing', () => {
      const store = mockStore(initialStoreState);
      const div = document.createElement('div');
      ReactDOM.render(<Provider store={store}><App fetchUser={mockFetchUserFn} /></Provider>, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('sends a request to fetch user upon mount', () => {
      expect(mockFetchUserFn).toHaveBeenCalled();
    });
  });
});
