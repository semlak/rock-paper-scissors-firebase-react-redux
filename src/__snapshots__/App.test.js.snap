// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components App shallow renders without crashing 1`] = `
<BrowserRouter
  basename=""
>
  <div
    className="App"
  >
    <Connect(AppNavbar) />
    <Connect(AuthenticationModal) />
    <div
      className="container main"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/app"
      />
    </div>
  </div>
</BrowserRouter>
`;
